### PR TITLE
Store shadow pass information for block entities in ShadowRenderer

### PR DESCRIPTION
**Pull request by IMS212**
IrisShaders/Iris#634 must be merged first.

### CODE REVIEW

File: src/main/java/me/jellysquid/mods/sodium/client/render/chunk/RenderSectionManager.java [hunk 1]
```diff
@@ -254,6 +257,7 @@ private void resetLists() {
         }
 
         this.visibleBlockEntities.clear();
+        ShadowRenderer.visibleBlockEntities.clear();
```
Comment (coderbot16) [inline src/main/java/me/jellysquid/mods/sodium/client/render/chunk/RenderSectionManager.java:260]: Is this needed? We're just assigning the list every frame anyways.

**Review by coderbot16 — CHANGES_REQUESTED** [review]
The way this PR sets up the visible block entities list is sorta weird. I think it would make more sense to just do it at the end of `RenderSectionManager::update` *if* we're in the shadow pass: https://github.com/IrisShaders/sodium-fabric/blob/a2d860985ca206d084547015a46e2362a66d200e/src/main/java/me/jellysquid/mods/sodium/client/render/chunk/RenderSectionManager.java#L165.

File: src/main/java/me/jellysquid/mods/sodium/client/render/chunk/RenderSectionManager.java (hunk 1 above)
Comment (IMS212) [reply src/main/java/me/jellysquid/mods/sodium/client/render/chunk/RenderSectionManager.java:260]: Sadly, trying to assign the list every frame seems to cause issues with vanilla/when shadows are disabled in my brief testing.

**Comment by IMS212** [comment]
Now for the Sodium side of the coin. Shouldn't the shadow block entities be added during the addBlockEntity method

**Comment by coderbot16** [comment]
Well, since the visibility list is only needed after setupTerrain (the update method is a part of that) completes, it makes more sense to just add them all at once with addAll instead of adding them one-by-one.

**Comment by IMS212** [comment]
#24 supersedes this.

**Pull request closed without merging**